### PR TITLE
Adding Mixtrack Pro FX switch

### PR DESCRIPTION
**Pull request by andybeetz**
The switch allows the FX encoder to control the meta knobs of Effect 1 on FX 1 and FX 2 (when shifted).

### CODE REVIEW

**Comment by andybeetz** [comment]
The discussion thread for this is available here [https://mixxx.discourse.group/t/modifying-the-mixtrack-pro-fx-mapping/27564](https://mixxx.discourse.group/t/modifying-the-mixtrack-pro-fx-mapping/27564).

File: res/controllers/Numark-Mixtrack-Pro-FX-scripts.js [hunk 1]
```diff
@@ -141,9 +146,23 @@ MixtrackProFX.EffectUnit = function(deckNumber) {
     this.effectParam = new components.Encoder({
         group: "[EffectRack1_EffectUnit" + deckNumber + "_Effect1]",
         shift: function() {
+            if(MixtrackProFX.fxControlsMeta)
+            {
```
Comment (ronso0) [inline res/controllers/Numark-Mixtrack-Pro-FX-scripts.js:150]: { should be on the same line, here and below
```suggestion
            if(MixtrackProFX.fxControlsMeta) {
```

**Review by ronso0 — CHANGES_REQUESTED** [review]

File: res/controllers/Numark-Mixtrack-Pro-FX-scripts.js (hunk 1 above)
Comment (andybeetz) [reply res/controllers/Numark-Mixtrack-Pro-FX-scripts.js:150]: Are those the only problems from pre-commit? I'm on Windows and pre-commit has failed to build the environment correctly so am unable to run it locally.

Comment (ronso0) [reply res/controllers/Numark-Mixtrack-Pro-FX-scripts.js:150]: no, looks like pre-commit did complain only about other issues, maybe only relevant on 2.4 with ES7 (IIRC).
I think you should rebase onto 2.3, foce-push, then check the pre-coomit result ("Details" link in the CI box). There's also a patch which you can apply manually.

> I'm on Windows and pre-commit has failed to build the environment correctly so am unable to run it locally.

If you don't find any help online, check into the Zulip development stream and search for pre-commit on Windows. IIRC this issue comes up frequently.
https://mixxx.zulipchat.com/

**Comment by andybeetz** [comment]
> Welcome and thank you! As a first-time contributor we need you to sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future.

I have signed the contributor agreement.

File: res/controllers/Numark-Mixtrack-Pro-FX-scripts.js [hunk 2]
```diff
@@ -141,9 +146,23 @@ MixtrackProFX.EffectUnit = function(deckNumber) {
...
+                this.group = "[EffectRack1_EffectUnit" + 2 + "_Effect1]"
```
Comment (Swiftb0y) [inline res/controllers/Numark-Mixtrack-Pro-FX-scripts.js:151]:
```suggestion
                this.group = "[EffectRack1_EffectUnit2_Effect1]";
```
same below please

**Pull request closed without merging**